File: day 6/majority_element_ii.py
from collections import Counter


class Solution:
    # Function to find the majority elements in the array
    def findMajority(self, arr):
        # Your Code goes here.
        ans = []
        n = len(arr)
        counts = dict(Counter(arr))

        for key, value in counts.items():
            if value > n / 3:
                ans.append(key)
        ans.sort()
        return ans


def main():
    t = int(input().strip())
    for _ in range(t):
        s = input().strip()
        nums = list(map(int, s.split()))
        ob = Solution()
        ans = ob.findMajority(nums)
        if not ans:
            print("[]")
        else:
            print(" ".join(map(str, ans)))


if __name__ == "__main__":
    main()

File: day 6/majority_element_ii.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
  public:
    // Function to find the majority elements in the array
    vector<int> findMajority(vector<int>& arr) {
        // Your code goes here.
        int n = arr.size();
        vector<int> ans;
        sort(arr.begin(), arr.end());
        
        map<int, int> hash;
        
        for(int i=0; i<n; i++){
            hash[arr[i]]++;
        }
        
        for(int i=0; i<hash.size; i++){
            if(hash[i] == n/3){
                ans.pushback(hash[i]    )
            }
        }
    }
};

int main() {
    int t;
    cin >> t;
    cin.ignore();
    while (t--) {

        string s;
        getline(cin, s);
        stringstream ss(s);
        vector<int> nums;
        int num;
        while (ss >> num) {
            nums.push_back(num);
        }
        Solution ob;
        vector<int> ans = ob.findMajority(nums);
        if (ans.empty()) {
            cout << "[]";
        } else {
            for (auto &i : ans)
                cout << i << " ";
        }
        cout << "\n";
    }
    return 0;
}
